modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -858,4 +917,41 @@ static boolean isRecoverable(Throwable throwable) {
             return false;
         }
     }
+
+    private void decrementRwTxCount(TxStateMeta txStateMeta) {
```

Comment:
As 've mentioned I'd consider moving it to index related modules. Within transactions scope it should be required to have extended txnStateMeta with tx type and catalog version.